gatsby/content/docs/free-small-matrix-server.mdx

```diff
@@ -0,0 +1,212 @@
+
+Make sure you can ssh to the server (using the ssh key you generated/picked)
+using the user 'ubuntu' and can use `sudo`.
+
+[sshkey]: https://docs.oracle.com/en/cloud/iaas/compute-iaas-cloud/stcsg/generating-ssh-key-pair.html
+
+## Open firewall
+
+### Oracle Cloud Security lists
+
+View resources -> Instances -> Select instance -> Virtual Cloud Network ->
+Public Subnet -> Security Lists -> Default -> Ingress
+
+Open incoming for CIDR 0.0.0.0/0:
+- 22/tcp for SSH (should be open already)
+- 80/tcp for HTTP
+- 443/tcp for HTTPS
+- 8448/tcp for Matrix federation
+- 3478/tcp, 5349/tcp, 3478/udp, 5349/udp, 49152-49172/udp for TURN/STUN (this is
+  used to get through firewalls or NAT for 1-on-1 audio/video calls and can be
+  omitted)
+
+### Ubuntu iptables
+
+The Oracle Cloud Ubuntu images come with somewhat restrictive iptables rules by
+default. Docker manages the instance firewall and we have the Oracle Cloud
+firewall in front, so let's remove the current firewall to avoid trouble:
+
+```sh
+apt purge netfilter-persistent iptables-persistent
+```
+
+## Remove useless stuff (optional)
+
+Oracle cloud includes a somewhat heavy monitoring daemon. We have better use for
+that memory since current versions of Synapse, the Matrix homeserver, can be
+memory hungry.
+
+```sh
+snap remove oracle-cloud-agent
+apt remove snapd open-iscsi lxd lxcfs
+```
+
+## Tune server (optional)
+
+I suggest enabling swap, since there's only 1 GB of RAM.
+
+```sh
+dd if=/dev/zero of=/swap bs=1M count=1k
+chmod 0600 /swap
+mkswap /swap
+swapon /swap
+echo '/swap none swap sw 0 0' >> /etc/fstab
+```
+
+## Point domain at server
+
+Assuming you're using a new domain only for this you need the following DNS
+records:
+
+- A record `$domain` pointing to `$instance_external_ip_address`
+- CNAME record `matrix.$domain` pointing to `$domain`
+- CNAME record `riot.$domain` pointing to `$domain`
+
+
+## Use matrix-docker-ansible-deploy
+
+Follow the [guide], with some tweaks:
+
+[guide]: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/README.md
+
+### Guide in a nutshell, TL;DR
+
+```sh
+git clone https://github.com/spantaleev/matrix-docker-ansible-deploy/
+cd matrix-docker-ansible-deploy
+mkdir inventory/host_vars/matrix.$domain
+cp examples/host-vars inventory/host_vars/matrix.$domain/vars.yaml
+cp examples/hosts inventory/hosts
+$EDITOR inventory/hosts
+$EDITOR inventory/host_vars/matrix.$domain/vars.yaml
+ansible-playbook -i inventory/hosts setup.yml --tags=setup-all
+ansible-playbook -i inventory/hosts setup.yml --tags=start
+ansible-playbook -i inventory/hosts setup.yml --tags=self-check
+ansible-playbook -i inventory/hosts setup.yml -e username=$user -e password=$pass -e admin=yes --tags=register-user
+```
+
+### Serve the base domain as well (unless you already have something serving it)
+
+```yaml
+matrix_nginx_proxy_base_domain_serving_enabled: true
+```
+
+### Disable all extras because of RAM
+
+```yaml
+matrix_dimension_enabled: false
```

Comment:
Dimension is disabled by default, because:
- it's not very useful, as mobile clients do not (yet) support custom integration managers.. and it doesn't have a responsive UI
- installing it is a bit involved (need to create a user and get an access token)